src/simulation/LoadGenerator.cpp

```diff
@@ -244,13 +291,46 @@ LoadGenerator::scheduleLoadGeneration(GeneratedLoadConfig cfg)
             MIN_UNIQUE_ACCOUNT_MULTIPLIER);
     }
 
-    if (cfg.mode == LoadGenMode::SOROBAN &&
+    if ((cfg.mode == LoadGenMode::SOROBAN_INVOKE ||
+         cfg.mode == LoadGenMode::SOROBAN_UPLOAD ||
+         cfg.mode == LoadGenMode::SOROBAN_INVOKE_SETUP) &&
         protocolVersionIsBefore(mApp.getLedgerManager()
                                     .getLastClosedLedgerHeader()
                                     .header.ledgerVersion,
                                 SOROBAN_PROTOCOL_VERSION))
     {
-        errorMsg = "Soroban mode requires protocol version 20 or higher";
+        errorMsg = "Soroban modes require protocol version 20 or higher";
+    }
+
+    if (cfg.mode == LoadGenMode::SOROBAN_INVOKE)
+    {
+        auto& sorobanCfg = mApp.getLedgerManager().getSorobanNetworkConfig();
+        if (mContractInstances.size() < cfg.nAccounts)
+        {
+            errorMsg = "must run SOROBAN_INVOKE_SETUP with at least nAccounts";
+        }
+        else if (cfg.nDataEntriesHigh > sorobanCfg.mTxMaxWriteLedgerEntries)
+        {
+            errorMsg = "nDataEntriesHigh larger than max write ledger entries";
+        }
+        // Wasm + instance + data entry reads
+        else if (cfg.nDataEntriesHigh + 2 > sorobanCfg.mTxMaxReadLedgerEntries)
```

Comment:
what are the magic numbers in these checks (2, 2'000)?

Reply:
The 2 is the Wasm + instance entry that we need to read, and the 2'000 is a rough estimate of wasm size. The comments are supposed to explain these magic numbers, but I can make them a little more clear if it's still not apparent what they are.